marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/provenance/dhProvLibTest.sjs

```diff
@@ -0,0 +1,107 @@
+
+function verifyProvenanceRecordCreatedAfterStartJob() {
+  const jobId = "verifyProvenanceRecordCreatedAfterStartJob-job-id-1";
+  const flowName = "verifyProvenanceRecordCreatedAfterStartJob-flow-1";
+  const jobStatus = "finished";
+  xdmp.invokeFunction(function () {
+    declareUpdate();
+    jobService.startJob(jobId, flowName);
+  });
+  const uris = queryProvenanceRecord(jobId);
+  let record = fn.head(fn.doc(uris.finalRecordUri));
+
+  return [
+    test.assertNotEqual(null, cts.doc(uris.stagingRecordUri)),
+    test.assertNotEqual(null, cts.doc(uris.finalRecordUri)),
+    test.assertEqual("job:verifyProvenanceRecordCreatedAfterStartJob-job-id-1", xs.string(fn.head(record.xpath("/*:document/*:activity/@*:id")))),
+    test.assertEqual("verifyProvenanceRecordCreatedAfterStartJob-job-id-1", xs.string(fn.head(record.xpath("/*:document/*:activity/*:label")))),
+    test.assertEqual("dh:Job", xs.string(fn.head(record.xpath("/*:document/*:activity/*:type")))),
+    test.assertNotEqual(null, xs.string(fn.head(record.xpath("/*:document/*:activity/*:startTime")))),
+    test.assertEqual(null, fn.head(record.xpath("/*:document/*:activity/*:endTime"))),
+    test.assertEqual("user:test-data-hub-developer", xs.string(fn.head(record.xpath("/*:document/*:agent/@*:id")))),
+    test.assertEqual("test-data-hub-developer", xs.string(fn.head(record.xpath("/*:document/*:agent/*:label")))),
+    test.assertEqual("dh:User", xs.string(fn.head(record.xpath("/*:document/*:agent/*:type")))),
+    test.assertEqual("job:verifyProvenanceRecordCreatedAfterStartJob-job-id-1", xs.string(fn.head(record.xpath("/*:document/*:wasAssociatedWith/*:activity/@*:ref")))),
+    test.assertEqual("user:test-data-hub-developer", xs.string(fn.head(record.xpath("/*:document/*:wasAssociatedWith/*:agent/@*:ref"))))
+  ];
+}
+
+function verifyProvenanceRecordCreatedAfterFinishJob() {
+  const jobId = "verifyProvenanceRecordCreatedAfterStartJob-job-id-2";
+  const flowName = "verifyProvenanceRecordCreatedAfterStartJob-flow-2";
+  const jobStatus = "finished";
+  xdmp.invokeFunction(function () {
+      declareUpdate();
+      jobService.startJob(jobId, flowName);
+      jobService.finishJob(jobId, jobStatus);
+    });
+  const uris = queryProvenanceRecord(jobId);
+  let record = fn.head(fn.doc(uris.finalRecordUri));
+
+  return [
+    test.assertNotEqual(null, cts.doc(uris.stagingRecordUri)),
+    test.assertNotEqual(null, cts.doc(uris.finalRecordUri)),
+    test.assertEqual("job:verifyProvenanceRecordCreatedAfterStartJob-job-id-2", xs.string(fn.head(record.xpath("/*:document/*:activity/@*:id")))),
+    test.assertEqual("verifyProvenanceRecordCreatedAfterStartJob-job-id-2", xs.string(fn.head(record.xpath("/*:document/*:activity/*:label")))),
+    test.assertNotEqual(null, xs.string(fn.head(record.xpath("/*:document/*:activity/*:startTime")))),
+    test.assertNotEqual(null, fn.head(record.xpath("/*:document/*:activity/*:endTime")))
+  ];
+}
+
+function queryProvenanceRecord(jobId) {
```

Comment:
I think it'd be better to put each of these test scenarios into a separate module, and then use setup.xqy to clear out the databases before each test. Then, instead of duplicating logic that's in the application code, you can safely grab the first document in the collection below in each database. 

See https://wiki.marklogic.com/display/ENGINEERING/Writing+Data+Hub+Tests#WritingDataHubTests-HowtoorganizeMLunittests for more information too on the downsides of multiple scenarios per test module.